astrolabe/spec_runner.py

```diff
@@ -213,17 +214,30 @@ def run(self, persist_cluster=False):
         # Step-5: interrupt driver workload and capture streams
         LOGGER.info("Stopping workload executor [PID: {}]".format(
             worker_subprocess.pid))
-        os.kill(worker_subprocess.pid, self.sigint)
-        stdout, stderr = worker_subprocess.communicate()
-        LOGGER.info("Stopped workload executor")
+        os.killpg(worker_subprocess.pid, self.sigint)
+        stdout, stderr = worker_subprocess.communicate(timeout=10)
+        LOGGER.info("Stopped workload executor [exit code: {}]".format(
+            worker_subprocess.returncode))
+        LOGGER.debug("stdout:".format(stdout))
+        LOGGER.debug("stderr:".format(stderr))
 
         # Stop the timer
         timer.stop()
 
         # Step-6: compute xunit entry.
         junit_test = junitparser.TestCase(self.id)
         junit_test.time = timer.elapsed
-        if worker_subprocess.returncode != 0:
```

Comment:
For some reason, on Ubuntu the worker_subprocess was returning a nonzero exit-code. Even after setting a `trap` for the `INT` signal, the return code was nonzero resulting in spurious failures.